### PR TITLE
[ROOT master] Get latest commits to adjust to new libs

### PR DESCRIPTION
**Pull request by mrodozov**
please test
lets see what needs to be adjusted

### CODE REVIEW

**Comment by smuzaffar** [comment]
did you try building locally?

**Comment by mrodozov** [comment]
now I'm trying it locally, after seeing how many things they've changed

**Comment by mrodozov** [comment]
please test

**Comment by mrodozov** [comment]
please test

**Comment by mrodozov** [comment]
please test

File: root.spec [hunk 1]
```diff
@@ -56,6 +57,7 @@ cmake ../%{n}-%{realversion} \
   -Dhdfs=OFF \
   -Dqt=OFF \
   -Dtmva=ON \
+  -DPython3_EXECUTABLE="${PYTHON3_ROOT}/bin/python3" \
```
Comment (smuzaffar) [inline root.spec:60]: why we are building with python3 here while CPYY_BACKEND_LIBRARY points to libcppyy_backend2_7.so?

Comment (mrodozov) [reply root.spec:60]: it uses both installations. the build fails if I don't provide the python3 executable location. reading the new cmake configs it seems the idea is to 'find out which python are available' 2, 3 or both and, use them nevertheless when I used only python2 it complained right away for incomplete cmake config in here:
https://github.com/root-project/root/blob/master/cmake/modules/SearchInstalledSoftware.cmake#L1490
In general I wanted to have a working build and then ask questions 'can we only use python2 if we want to and how'

Comment (smuzaffar) [reply root.spec:60]: ah ok, I thought it was only building for python3. If it is build for both then this looks good. 
Can we build yoda also for both python version?

Comment (mrodozov) [reply root.spec:60]: I changed the CPYY_BACKEND_LIBRARY to libcppyy_backend3_8 
and few other things to try. We can see also if pyroot builds with py3 (which is available for py3 and should work)

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
please test

**Comment by smuzaffar** [comment]
closing in favor of https://github.com/cms-sw/cmsdist/pull/5724

**Pull request closed without merging**